dumpling/export/dump.go

```diff
@@ -81,10 +81,17 @@ func NewDumper(ctx context.Context, conf *Config) (*Dumper, error) {
 		selectTiDBTableRegionFunc: selectTiDBTableRegion,
 	}
 
+	var err error
+
 	d.metrics = newMetrics(conf.PromFactory, conf.Labels)
 	d.metrics.registerTo(conf.PromRegistry)
+	defer func() {
+		if err != nil {
+			d.metrics.unregisterFrom(conf.PromRegistry)
```

Comment:
if dump success, will metrics unregisterFrom? 
dm may do nothing when stop dumpling, and resume by new dumper again, i'm just afraid it may causes duplicate  registration.

Reply:
it's in d.Close, caller should not forget it

Reply:
ok, we will cancel ctx when pause [here](https://github.com/pingcap/tiflow/blob/master/dm/dm/worker/subtask.go#L606), i think i will be ok now.